source/extensions/filters/http/cache/cache_headers_utils.cc

```diff
@@ -287,52 +303,40 @@ constexpr absl::string_view header_separator = "\n";
 constexpr absl::string_view in_value_separator = "\r";
 }; // namespace
 
-std::string VaryHeader::createVaryKey(const Http::HeaderMap::GetResult& vary_header,
-                                      const Http::RequestHeaderMap& entry_headers) {
-  if (vary_header.empty()) {
-    return "";
+absl::optional<std::string>
+VaryHeader::createVaryIdentifier(const absl::btree_set<absl::string_view>& vary_header_values,
+                                 const Http::RequestHeaderMap& request_headers) const {
+  std::string vary_identifier = "vary-id\n";
```

Comment:
Should this be a constexpr?

Reply:
The issue here is that the string is dynamically allocated on every invocation of `createVaryIdentifier`; moving it outside of the method and making it a constexpr is an easy win.

Reply:
I'm not a C++ expert, so forgive me for being dense.

If I move this out to be a constexpr string_view, then I either have to copy the string_view to a string (and pay the allocation anyway) or I have to not use absl::StrAppend later on, and pay for additional copies there. Is there another way that constexpr string_view saves me allocations that I am not considering?

Reply:
Yeah vary_identifier is being appended to in most cases. The only opportunity I see to improve this is to avoid the alloc in the case below where absl::nullopt is being returned.

You could do a pass over all the headers first and if any of them are not allowed return nullopt before trying to accumulate any strings. Not sure if that's worth it.